src/hooks/useLogs.ts

```diff
@@ -0,0 +1,230 @@
+import { useEffect, useMemo } from "react";
+import { set } from "unchanged";
+import { useTrackedEffect } from "ahooks";
+import { useSubscription } from "urql";
+
+import equals from "@/utils/helpers/equals";
+import useQuery from "@/hooks/useQuery";
+import type {
+  Request_Event_Logs_Order_By,
+  Request_Logs,
+} from "@/graphql/generated";
+
+import type { SubscriptionHandler } from "urql";
+
+const defaultFields = `
+  id
+  created_at
+  updated_at
+  request_id
+  start_time
+  end_time
+  duration
+  path
+  user {
+    display_name
+  }
+  datasource {
+    name  
+  }
+`;
+
+const currentLogQuery = `
+  query ($id: uuid!) {
+    request_logs_by_pk (id: $id) {
+      ${defaultFields}
+      request_event_logs(order_by: {timestamp: desc}) {
+        id
+        duration
+        event
+        path
+        query
+        query_key
+        query_sql
+        query_key_md5
+        queue_prefix
+        time_in_queue
+        timestamp
+        error
+      }
+    }
+  }
+`;
+
+const allLogsQuery = `
+  query ($offset: Int, $limit: Int, $where: request_logs_bool_exp, $order_by: [request_logs_order_by!]) {
+    request_logs (offset: $offset, limit: $limit, where: $where, order_by: $order_by) {
+      ${defaultFields}
+      request_event_logs(order_by: {timestamp: desc}) {
+        path
+        error
+      }
+      request_event_logs_aggregate {
+        aggregate {
+          count
+        }
+      }
+    }
+    request_logs_aggregate (where: $where) {
+      aggregate {
+        count
+      }
+    }
+  }
+`;
+
+const allLogsSubscription = `
+  subscription ($offset: Int, $limit: Int, $where: request_logs_bool_exp, $order_by: [request_logs_order_by!]) {
+    request_logs (offset: $offset, limit: $limit, where: $where, order_by: $order_by) {
+      id
+    }
+  }
+`;
+
```

Comment:
all queries should be provided via urql and graphql-codegen, please check other `.gql` files and components with queries

Reply:
Example of queries https://github.com/mlcraft-io/client-v2/commit/94f0729f78f6ced1afb8f5e9c4de2d86d6545501#diff-a3bcf9b962d5e633c30ddf2d6ee0ee63b3e5c1fff208828df74fa5a501ed1c61